data/org.freedesktop.portal.Secret.xml

```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0"?>
+<!--
+ Copyright (C) 2019 Red Hat, Inc.
+
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library. If not, see <http://www.gnu.org/licenses/>.
+
+ Author: Daiki Ueno <dueno@redhat.com>
+-->
+
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <!--
+      org.freedesktop.portal.Secret:
+      @short_description: Portal for retrieving application secret
+
+      The Secret portal allows sandboxed applications to retrieve a
+      per-application secret.  The secret can then be used for
+      encrypting confidential data inside the sandbox.
+
+      This documentation describes version 1 of this interface.
+  -->
+  <interface name="org.freedesktop.portal.Secret">
+
+    <!--
+        RetrieveSecret:
+        @fd: Writable file descriptor for transporting the secret
+        @options: Vardict with optional further information
+        @handle: Object path for the #org.freedesktop.portal.Request object representing this call
+
+        Retrieves a master secret for a sandboxed application.
+
+        Supported keys in the @options vardict include:
+        <variablelist>
+          <varlistentry>
+            <term>token s</term>
+            <listitem><para>
+	      An opaque string returned by a previous org.freedesktop.portal.Secret.RetrieveSecret() call.
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+    -->
+    <method name="RetrieveSecret">
+      <annotation name="org.gtk.GDBus.C.Name" value="retrieve_secret"/>
+      <annotation name="org.gtk.GDBus.C.UnixFD" value="true"/>
+      <arg type="h" name="fd" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="handle" direction="out"/>
+    </method>
```

Comment:
Given that there is no user interaction here (we never show a dialog), this could probably be simplified by omitting the reponse object, and just making this a plain method call. The secret will be returned via the fd anyway.

Reply:
There is actually an interaction: if the login keyring is locked, the dialog will be shown. However, the situation is exceptional and it might make more sense to to just return error in that case.

Reply:
ah, ok. in that case, lets leave things as-is.